### PR TITLE
ddl: make sequence cache-opt mutually exclusive to each other

### PR DESCRIPTION
**Pull request by AilinKid**
Signed-off-by: ailinkid <314806019@qq.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/31265

### What is changed and how it works?
1: sequence cachevalue(int) and cache(bool) opt should be mutually exclusive to each other when setting or altering.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

```release-note
None
```

### CODE REVIEW

File: ddl/sequence_test.go [hunk 1]
```diff
@@ -1088,3 +1090,17 @@ func (s *testSequenceSuite) TestAlterSequencePrivilege(c *C) {
 	c.Assert(err.Error(), Equals, "[planner:1142]ALTER command denied to user 'myuser'@'localhost' for table 'my_seq'")
 	tk.MustExec("drop sequence if exists my_seq")
 }
+
+func TestDdl_AlterSequenceIssue31265(t *testing.T) {
```
Comment (djshow832) [inline ddl/sequence_test.go:1094]: This case is different from the issue. Try to set it no-cache and then set the cache size of it. The result should be cached.

File: ddl/sequence_test.go [hunk 2]
```diff
@@ -1088,3 +1090,17 @@ func (s *testSequenceSuite) TestAlterSequencePrivilege(c *C) {
...
+	store, clean := testkit2.CreateMockStore(t)
+	defer clean()
+
+	tk := testkit2.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("create sequence seq cache=1 nocache")
```
Comment (djshow832) [inline ddl/sequence_test.go:1100]: `cache=1` conflicts with `nocache`, why not report an error?

Comment (AilinKid) [reply ddl/sequence_test.go:1100]: the late one will take effect

**Review by djshow832 — APPROVED** [review]

**Pull request merged**